home_work_8/index.js

```diff
@@ -0,0 +1,188 @@
+  const lengthOfArray = +prompt(
+    "Enter the length of the array as a number (from 1 to 10)"
+  );
+  if (isNaN(lengthOfArray) || lengthOfArray > 10) {
+    i--;
+  } else {
+    for (let i = 1; i <= lengthOfArray; i++) {
+      const enteredValue = +prompt(`Enter a numeric value №${i}`);
+      if (!isNaN(enteredValue)) {
+        testArray.push(enteredValue);
+      } else {
+        i--;
+      }
+    }
+  }
+}
+
+console.log(testArray);
+const sortedTestArray = testArray.sort((a, b) => a - b);
+
+console.log(sortedTestArray);
+
+/*
+3.0 Дано масив - список покупок. Кожен елемент масиву - це обʼєкт вигляду:
+{productName: 'bread', productPrice: 14.5, productQuantity: 2}.
+Мінімальний довжина такого масиву - 6
+*/
+
+const productsList = [
+  { productName: "bread", productPrice: 14.5, productQuantity: 2 },
+  { productName: "salo", productPrice: 120.5, productQuantity: 4 },
+  { productName: "garlic", productPrice: 60.5, productQuantity: 13 },
+  { productName: "carrots", productPrice: 14.5, productQuantity: 22 },
+  { productName: "cabbage", productPrice: 12.5, productQuantity: 44 },
+  { productName: "potatoes", productPrice: 6, productQuantity: 56 },
+];
+
+// 3.1 Порахувати загальну вартість корзини та вивести суму у зрозумілому форматі
+const totalProductPrice = productsList.reduce(
+  (sum, price) => sum + price.productPrice * price.productQuantity,
+  0
+);
+console.log(`Total cost: ${totalProductPrice}`);
+
+// 3.2 Порахувати найменшу кількість продукту, який потрібно купити
+const minQuantity = productsList.sort(
+  (el1, el2) => el1.productQuantity - el2.productQuantity
+)[0];
+console.log(
+  `Least of all ${minQuantity.productName}: ${minQuantity.productQuantity} pieces`
+);
+
+// 3.3 Порахувати загальну кількість продуктів
+const totalProductQuantity = productsList.reduce(
+  (sum, quantity) => sum + quantity.productQuantity,
+  0
+);
+console.log(`Total products: ${totalProductQuantity}`);
+
+// 3.4 Знайти найдорожчий продукт
+const mostExpensiveProduct = productsList.sort(
+  (el1, el2) => el2.productPrice - el1.productPrice
+)[0];
+console.log(
+  `Most expensive: ${mostExpensiveProduct.productName}, it costs ${mostExpensiveProduct.productPrice}`
+);
+
+// 3.5 Створити функцію, яка повинна додавати новий продукт в існуючий масив
+const createNewProduct = () => {
+  const NEW_PRODUCT = {
+    productName: "",
+    productPrice: 0,
+    productQuantity: 0,
+  };
+  for (const [key] of Object.entries(NEW_PRODUCT)) {
+    const value = prompt([key]);
+    if (!validateValue(value) && key === "productName") {
+      NEW_PRODUCT[key] = EMPTY_VALUE;
+    } else if (key === "productPrice" || key === "productQuantity") {
+      NEW_PRODUCT[key] = Number(value) ? Number(value) : 0;
+    } else {
+      NEW_PRODUCT[key] = value;
+    }
+  }
+  productsList.push(NEW_PRODUCT);
+  return productsList;
+};
+console.log(createNewProduct());
+
+console.log(productsList);
+
+/*
+  for (let key of NEW_PRODUCT) {
```

Comment:
You need a `for in` loop, `for of` its for arrays and other iterable objects(string or map)

Reply:
thx, updated